src/action/SummationAction.cpp

```diff
@@ -0,0 +1,44 @@
+#include "SummationAction.hpp"
+#include "mesh/Data.hpp"
+#include "mesh/Edge.hpp"
+#include "mesh/Mesh.hpp"
+
+namespace precice {
+namespace action{
+
+SummationAction::SummationAction(
+	Timing				timing,
+	std::vector<int>	sourceDataID,
+	int					targetDataID,
+	const mesh::PtrMesh	&mesh)
+	: Action(timing, mesh, mapping::Mapping::MeshRequirement::VERTEX), _targetData(mesh->data(targetDataID))
+{
+
+	for(auto sourceID : sourceDataID){
+		_sourceData.push_back(mesh->data(sourceID));
+	}
+
+	for(const auto & source : _sourceData) {
+		PRECICE_ASSERT(source->getDimensions() == _targetData->getDimensions(), source->getDimensions(), _targetData->getDimensions());
+	}
+
+}
+
+void SummationAction::performAction(
+	double time,
+	double dt,
+	double computedPartFullDt,
+	double fullDt)
+{
+	PRECICE_TRACE();
+	auto & targetValues = _targetData->values();
+	auto sum = _sourceData.at(0)->values();
+
+	for(int i = 1; i < _sourceData.size(); ++i){
+		sum += _sourceData.at(i)->values();
+	}
+	targetValues = sum;
```

Comment:
Mmh, don't we need to loop over the complete mesh? Similarly to [here](https://github.com/precice/precice/blob/develop/src/action/ScaleByDtAction.cpp#L43-L45). Or is the "+=" here overloaded?

Reply:
When i checked, it was overloaded for Eigen vectors and matrices but I changed it to the loop version anyways since it is not obvious.